content/en/stateless_apps.md

```diff
@@ -6,12 +6,16 @@ category: technology
 
 ## What it is
 
-A stateless application doesn’t save any client session (state) data on the server where the application lives. Each session is carried out as if it was the first time and responses are not dependent upon data from a previous session and provides functionality to use print services, CDN (Content Delivery Network) or the Web Servers in order to process every short-term request. For example, someone is searching a question in the search engine and pressed the Enter button. In case if the searching operation gets interrupted or closed due to some reason, you have to start a new one as there is no saved data for your previous request.
+When we speak of stateless (and [stateful](https://glossary.cncf.io/stateful_apps/)) apps, state refers to any data the app needs to store to function as designed. Stateless apps, as the name suggests, neither read nor store state information. You already use stateless applications on a regular basis. When you use a search engine to pull up a query about a question you have, you’re using a stateless application. Even if you’re interrupted during the process, you can come straight back and ask the same question, or a different question, because all memory of the previous transaction is gone between takes. 
+
+
 
 ## Problem it addresses
 
-Stateless applications tackle the problem of resiliency, because different pods across a [cluster](/cluster/) can work independently, with multiple requests coming to them at the same time. If there’s a problem, you can easily restart the application, and it will return to its initial state with little or no downtime. As such, the benefits of stateless applications include resiliency, elasticity, and high availability. However, most applications we use today are at least partly [stateful](/stateful_apps/), as they store things like preferences and settings to improve the user experience.
+When it comes to stateful applications it has to store the details of every transaction that happened in the previous step. So the need for extensive memory in the form of databases and servers can make stateful applications  slower, heavier, harder to manage. 
```

Comment:
I think we're on the wrong track to say stateless apps solve the problem of stateful apps. Instead we look at both stateful and stateless apps as solving the problem of how do we isolate app development work to a specific domain. The point being, we don't adopt stateless apps because stateful apps are hard. We adopt stateless apps so that our devs can focus on writing app logic/features and not worry about storing state, instead delegating the task to another service. The stateless/stateful paradigm is to some degree a natural consequence of writing micro services.